cpp/benchmarks/replay/replay.cpp

```diff
@@ -97,6 +98,50 @@ struct allocation {
   std::size_t size{};
 };
 
+/**
+ * @brief Partial polyfill for C++20 std::barrier
```

Comment:
@wence- Would it help if we moved RMM to build with C++20? I think that should be okay now that we require CUDA 12+. cuDF has already made that transition.

Maybe we need to transition more of RAPIDS first since much of RMM is still header-only (and probably needs to remain C++17 compatible until the rest of RAPIDS is on C++20), but we could _at least_ build RMM benchmarks with C++20 to reduce the code required here.

Reply:
Done.